# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Zuhairsyed123/ML_internship_Assignments/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [2]:
### Selected Lane: Refresh / Content Opportunity Scoring

#I selected the Refresh / Content Opportunity Scoring lane because it focuses on a practical business problem: identifying which content pages should be reviewed first. Large content inventories make manual review difficult, so a ranked recommendation system can help prioritize pages that show signs of declining performance or growth opportunity. The starter project already demonstrates that machine learning can improve prioritization, making this lane a strong choice for further investigation during the internship.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question

Which content pages should be reviewed first for refresh, expansion, protection, or monitoring based on observable search and engagement signals?

### Decision Being Improved

The decision of which pages should receive attention first when content review resources are limited.

### Action

Content managers or SEO teams can review, update, expand, optimize, or monitor the highest-priority pages.

### Cost of a Wrong Recommendation

If a page is incorrectly prioritized, time and resources may be spent on content that does not need improvement. If an important page is missed, potential traffic and engagement opportunities may be lost.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

total_pages = len(df)
declining_pages = (df["trend_direction"] == "down").sum()
decline_rate = (declining_pages / total_pages) * 100

print("Total rows:", total_pages)
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average sessions (90d):", round(df["sessions_90d"].mean(), 2))
print("Declining pages:", declining_pages)
print("Decline rate (%):", round(decline_rate, 2))

Total rows: 30000
Average impressions (90d): 5200.37
Average sessions (90d): 37.07
Declining pages: 16262
Decline rate (%): 54.21


## Dataset Observations

The starter dataset contains 30,000 content pages and 44 features related to search visibility, engagement, content characteristics, and performance trends. The average page received 5,200.37 impressions and 37.07 sessions over the previous 90 days. Additionally, 16,262 pages are classified as declining based on the provided trend label.

These numbers indicate that content decline is a common pattern within the dataset and that there is sufficient search visibility and engagement data available for analysis. Therefore, identifying which pages should be reviewed first for refresh or optimization is a meaningful problem and makes the Refresh / Content Opportunity Scoring lane a strong choice for the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will provide decision-support recommendations based on observed search and engagement signals. By analyzing patterns in impressions, sessions, CTR, content age, and performance trends, it can help identify pages that may deserve review for refresh, expansion, protection, or monitoring.

The results will be observational and directional rather than causal. A high score suggests that a page may be a good candidate for review, but it does not guarantee that updating the page will improve performance.

This project cannot prove that a content refresh causes traffic growth, cannot determine Google's ranking algorithms, and cannot predict future rankings with certainty. All recommendations should be reviewed by humans before action is taken.

In [8]:
claim_type = [
    "observed",
    "directional",
    "decision-support"
]

print("Claims supported by this project:")
for c in claim_type:
    print("-", c)

Claims supported by this project:
- observed
- directional
- decision-support


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.